### ISSUE TITLE
"LESS: Unexpected token u" error when compiling LESS file with empty result

### ISSUE THREAD

**Issue by NigelWhatling** [madskristensen/WebEssentials2013#1696]
_This may be related to [Issue#1170](https://github.com/madskristensen/WebEssentials2013/issues/1170) (which according to the comments has been fixed)._

Recently updated to v2.5.3. When compiling a LESS file with no resulting CSS, a "LESS: Unexpected token u" error is given. I ran across this when making a change to the variables.less file of bootstrap. The file only contains variable definitions and is imported into other LESS files. This file no longer compiles.

As with [Issue#1170](https://github.com/madskristensen/WebEssentials2013/issues/1170), adding some element code (e.g. `body { cursor:auto; }`) to the file corrects the issue.

I have also found that adding an 'important' comment (e.g. `/* bug fix comment */`) fixed the issue (I'm guessing because these comments are retained in the output).

I'm working around the issue with the comment for now. But it would good to fix.

**Comment by rosslynquynhnguyen** [comment]
This is still a problem for me. I am using Web Essentials 2013 for Update 4.

The workaround I use for now is to add @charset "utf-8";

**Comment by rjgotten** [comment]
Go to `%USERPROFILE%\AppData\Local\Microsoft\VisualStudio\12.0\Extensions` which is the folder where per-user Visual Studio extensions reside. WebEssentials will be located in a subfolder with a randomly generated name.

From inside the WebEssentials folder, open up the file `Resources\nodejs\tools\server\services\srv-less.js` and go to line 65, which reads:
```js
map = JSON.parse(output.map);
```
The problem is source map output may be the `undefined` value. `JSON.parse` can only parse strings, so it casts that to the string value `"undefined"` before parsing, but `JSON` does not recognize that as valid token. (It only understands the null value, not the undefined value.)

So... change line 65 to read:
```js
map = JSON.parse(output.map || "null");
```
And voilà; LESS compilation on files with empty output works again.

**Comment by CheloXL** [comment]
@rjgotten Thanks!!! That was driving me crazy!

**Comment by figalex** [comment]
@rjgotten That fixed it! Thanks!

**Comment by Joebeazelman** [comment]
Is there a reason why this fix isn't included in the latest updates?  Surely, if I was the developer I would want to include it the fix.

**Comment by rjgotten** [comment]
> Is this the current state of software?

No, it's just Web Essentials falling flat on its face every other release due to a chronic shortage of proper unit tests on Visual Studio extension with a scope wide enough that it has begun to follow the god class anti-pattern.

Web Essentials should be torn apart into smaller modules that can be invididually tested and individually updated. It's been mentioned before a few times...

**Comment by gligoran** [comment]
Thanks you @rjgotten, this fixed it for me too.

I hope the Web Essentials team (@madskristensen) implements this fix with the next update.

P.S. If anyone in the future is wondering, a restart of Visual Studio is required. At least for me it was.

**Comment by MCluck90** [comment]
:+1: @rjgotten's solution worked for me. This should really be implemented in the next update.

**Comment by behrangsa** [comment]
:+1: Looking forward for this fix!

**Comment by nbarbettini** [comment]
@rjgotten Thanks for the fix. This was a a really confusing error, but it's working fine thanks to you.

Hoping this is fixed permanently in a future release!

**Comment by behrangsa** [comment]
@rjgotten :+1:

**Comment by RichardJECooke** [comment]
Why is this still not in the release version months later? :-(

**Comment by niklasHagner** [comment]
Unbelievable that this still hasn't been fixed.
I've had to include the fix (chagning a line in to srv-less.js as suggested by @rjgotten ) as a part of the developer documentation for a project at my company.
